### PR TITLE
new project metadata representation component

### LINKED ISSUE #2276: Missing Project metadata representation component
**Describe the bug**
Not sure why, but there isn't a metadata representation component specific for projects available at the submission form, like there is for persons or org-units entity types.
And for that reason, when trying to associate a Project entity to a Publication, for that new relation, no label is shown.
image

instead a default component is used for that specific relationship `PlainTextMetadataListElementComponent` that only applies to entities of type `Publication`
image

**To Reproduce**
1. Using DSpace Angular 7.5 (I recall this didn't happen in an earlier version)
2. Using Open AIRE Publication Submission Form
3. Associate an entity of type Project to the Publication
4. And empty label appears

**Expected behavior**
In earlier version I recall that it was used virtual metadata to display data. But now, the component is hooked to a type. So there is a missing Project component.

### PR DESCRIPTION
**Pull request by paulo-graca**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #2276

## Description
This PR adds a new component to address issue #2276.

## Instructions for Reviewers
You will find that when trying to create a relationship from a Project to a Publication, the project label is shown on the submission form (the dc:title field)
image

### CODE REVIEW

**Comment by artlowel** [comment]
@paulo-graca The code looks sensible, but I don't seem to be able to reproduce the issue in the first place. If I create a relationship to a Project in the submission it already has a proper name:
```xml
            <row>
                <relation-field>
                    <relationship-type>isProjectOfPublication</relationship-type>
                    <search-configuration>project</search-configuration>
                    <label>Project</label>
                    <hint>import a project</hint>
                </relation-field>
            </row>
```

Perhaps you can send me the exact submission config I need to reproduce the issue?

**Comment by paulo-graca** [comment]
Thank you @artlowel for your feedback. I will try to gather more data.

**Comment by paulo-graca** [comment]
@artlowel I'm using `openAIREPublicationSubmission`

With the field specification:
```
            <row>
                <relation-field>
                    <relationship-type>isProjectOfPublication</relationship-type>
                    <search-configuration>project</search-configuration>
                    <repeatable>true</repeatable>
                    <name-variants>false</name-variants>
                    <label>Funding</label>
                    <hint>Add a related Funding</hint>
                    <linked-metadata-field>
                        <dc-schema>dc</dc-schema>
                        <dc-element>relation</dc-element>
                        <input-type>onebox</input-type>
                    </linked-metadata-field>
                    <externalsources>openAIREFunding</externalsources>
                    <required></required>
                </relation-field>
            </row>
```
and the relationship type definition has the following:
```
	<type>
		<leftType>Publication</leftType>
		<rightType>Project</rightType>
		<leftwardType>isProjectOfPublication</leftwardType>
		<rightwardType>isPublicationOfProject</rightwardType>
		<leftCardinality>
			<min>0</min>
		</leftCardinality>
		<rightCardinality>
			<min>0</min>
		</rightCardinality>
	</type>
```

This is what's is happening:

invisible-project

File: src/app/entity-groups/research-entities/metadata-representations/project/project-item-metadata-list-element.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  <span class="text-muted">
+      <span class="item-list-project-title">
+        <span [innerHTML]="mdRepresentation.firstMetadataValue(['dc.title'])"></span>
+      </span>
+  </span>
+</ng-template>
+<ds-truncatable [id]="mdRepresentation.id">
+  <a [routerLink]="[itemPageRoute]"
+     [innerHTML]="mdRepresentation.getValue()"
+     [ngbTooltip]="mdRepresentation.allMetadata(['dc.title']).length > 0 ? descTemplate : null"></a>
```
Comment (artlowel) [inline src/app/entity-groups/research-entities/metadata-representations/project/project-item-metadata-list-element.component.html:11]: Same comment

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @paulo-graca!

The key was to use [the openaire4 virtual metadata mapping](https://github.com/DSpace/DSpace/blob/main/dspace/config/spring/api/virtual-metadata.xml.openaire4). If I use that, I can reproduce the issue.

However your fix doesn't work for me in all cases, because you've created a new component with a decorator specific to dspace (i.e. `@metadataRepresentationComponent`). You need to add that component to the [ENTRY_COMPONENTS](https://github.com/DSpace/dspace-angular/blob/d56497edfa26f65fd11e78e3193a6812758a1cd9/src/app/entity-groups/research-entities/research-entities.module.ts#L35-L61) array of its module. That way angular will know it exists and process that decorator when the app starts. Otherwise you just have to be lucky that it's already been processed by the time you need it

**Comment by paulo-graca** [comment]
Thank you @artlowel for your feedback, I will address it soon as possible!

File: src/app/entity-groups/research-entities/metadata-representations/project/project-item-metadata-list-element.component.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<ng-template #descTemplate>
+  <span class="text-muted">
+      <span class="item-list-project-title">
+        <span [innerHTML]="mdRepresentation.firstMetadataValue(['dc.title'])"></span>
```
[the remark this answers is not in the document]
Comment (paulo-graca) [reply src/app/entity-groups/research-entities/metadata-representations/project/project-item-metadata-list-element.component.html]: I've replaced it with `dsoNameService.getName(mdRepresentation)`

File: src/app/entity-groups/research-entities/metadata-representations/project/project-item-metadata-list-element.component.html (hunk 1 above)
Comment (paulo-graca) [reply src/app/entity-groups/research-entities/metadata-representations/project/project-item-metadata-list-element.component.html]: I've replaced it with `dsoNameService.getName(mdRepresentation)`

**Comment by paulo-graca** [comment]
@artlowel I've just addressed your feedback. It's working for me.
image

I will just address the remaining styling issues.

**Comment by paulo-graca** [comment]
I don't have a test system setup on my remote host, I was trying to do that. But it will take me some time.
I didn't saw your feedback in time before my last push, but I think I was in the right track. Thank you! I will try to address it.

**Pull request merged**